From ade6ee5eab8d05a4abe4c57cd162119c5a4226aa Mon Sep 17 00:00:00 2001
From: Evan Wang <xswang@marvell.com>
Date: Tue, 11 Apr 2017 17:24:00 +0800
Subject: [PATCH 0954/1345] fix: comphy: a3700: correct the unmap address
 during USB3 phy power on using lane 2

commit  40da6849ede961136938371b010b9ab651124ba0 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- There is an error when unmap USB3 comphy rousource.
- The patch fix the error.

Change-Id: Icf9342b192166bbcefbfa68f4b972ae5a294ecd8
Signed-off-by: Evan Wang <xswang@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/38653
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/phy/phy-comphy-a3700.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/phy/phy-comphy-a3700.c b/drivers/phy/phy-comphy-a3700.c
index 61b8fb8..d7a202d 100644
--- a/drivers/phy/phy-comphy-a3700.c
+++ b/drivers/phy/phy-comphy-a3700.c
@@ -719,7 +719,8 @@ static int mvebu_a3700_comphy_usb3_power_on(struct mvebu_comphy_priv *priv,
 	/* Unmap resource */
 	if (comphy->index == COMPHY_LANE2) {
 		devm_iounmap(&pdev->dev, comphy_indir_regs);
-		devm_release_mem_region(&pdev->dev, res->start, resource_size(res_indirect));
+		devm_release_mem_region(&pdev->dev, res_indirect->start,
+					resource_size(res_indirect));
 	} else {
 		devm_iounmap(&pdev->dev, usb3_gbe1_phy_regs);
 		devm_release_mem_region(&pdev->dev, res->start, resource_size(res));
-- 
1.7.9.5

